safehttp/response_writer.go

```diff
@@ -38,7 +40,12 @@ type ResponseWriter interface {
 	//
 	// If the ResponseWriter has already been written to, then this method panics.
 	Redirect(r *IncomingRequest, url string, code StatusCode) Result
+}
 
+// ResponseHeadersWriter is used to alter the HTTP response headers.
+//
+// A ResponseHeadersWriter may not be used after the Handler.ServeHTTP method has returned.
+type ResponseHeadersWriter interface {
```

Comment:
I would document that if somethings has already been written this entire type methods set becomes a no-op.

Reply:
Good idea!

Reply:
Actually, I'm going to think a bit about documenting it better with regards to lifetime. I think that nothing should hold a reference to `ResponseHeadersWriter` after the `Commit` or `OnError` function has been exited.